pkg/provisioningv2/rke2/planner/encryptionkeyrotation.go

```diff
@@ -447,17 +447,52 @@ func (p *Planner) encryptionKeyRotationRestartService(cp *rkev1.RKEControlPlane,
 			encryptionKeyRotationSecretsEncryptStatusOneTimeInstruction(cp),
 		)
 	}
+
+	capiCluster, err := rke2.GetOwnerCAPICluster(cp, p.capiClusters)
+	if err != nil {
+		return "", status, err
+	}
+
+	plan, err := p.store.Load(capiCluster, cp)
+	if err != nil {
+		return "", status, err
+	}
+
+	joinServer, err := p.electInitNode(cp, plan)
+	if err != nil {
+		return "", status, err
+	}
+
+	tokensSecret, err := p.generateSecrets(cp)
+	if err != nil {
+		return "", status, err
+	}
+
+	_, config, err := p.generatePlanWithConfigFiles(cp, tokensSecret, entry, joinServer)
+	if err != nil {
+		return "", status, err
+	}
+
+	probes, err := p.generateProbes(cp, entry, config)
+	if err != nil {
+		return "", status, err
+	}
+	nodePlan.Probes = probes
```

Comment:
Is there any way we can refactor probe generation to not require this giant preamble every time we need the probes for a node...?